modules/representation_management/app/models/representation_management/next_steps_email_data.rb

```diff
@@ -20,13 +19,84 @@ class NextStepsEmailData
     validates :first_name, presence: true
     validates :form_name, presence: true
     validates :form_number, presence: true
-    validates :representative_type, presence: true
-    validates :representative_type, inclusion: { in: AccreditedIndividual.individual_types.keys }
-    validates :representative_name, presence: true
-    validates :representative_address, presence: true
+    validates :entity_type, presence: true
+    validates :entity_id, presence: true
+    validate :entity_exists?
 
-    def representative_type_humanized
-      @representative_type_humanized ||= representative_type.humanize.titleize
+    def entity
+      @entity ||= find_entity
+    end
+
+    def entity_display_type
+      display_type =
+        if entity.is_a?(Veteran::Service::Representative)
+          veteran_service_representative_type
+        elsif entity.is_a?(AccreditedIndividual)
+          entity.individual_type
+        else
+          'Veteran Service Organization'
+        end
+      display_type.humanize.titleize
+    end
+
+    def entity_name
+      if entity_type == 'individual'
+        entity.full_name.strip
+      elsif entity_type == 'organization'
+        entity.name.strip
+      end
+    end
+
+    def entity_address
+      <<~ADDRESS.squish
+        #{entity.address_line1}
+        #{entity.address_line2}
+        #{entity.address_line3}
+        #{entity.city}, #{entity.state_code} #{entity.zip_code}
+        #{entity.country_code_iso3}
+      ADDRESS
+    end
+
+    private
+
+    def find_entity
+      if entity_type == 'individual'
+        find_representative
+      elsif entity_type == 'organization'
+        find_organization
+      end
+    end
+
+    def entity_exists?
+      return unless entity.nil?
+
+      errors.add(:entity, 'Entity not found')
+    end
+
+    def find_representative
+      AccreditedIndividual.find_by(id: entity_id) ||
+        Veteran::Service::Representative.find_by(representative_id: entity_id)
+    end
+
+    def find_organization
+      AccreditedOrganization.find_by(id: entity_id) ||
+        Veteran::Service::Organization.find_by(poa: entity_id)
+    end
+
+    def veteran_service_representative_type
+      representative_type = entity.user_types.first
```

Comment:
I need to riff on this with you for a little bit.

How can we confirm that the first type in the `user_types` array matches what the vet expects the representative's role to be?

For example, if `user_types` is set to `['claims_agent', 'attorney']` and the vet intends to assign an attorney, it could be confusing if we inform them that the representative is a claims agent.

@jvcAdHoc Do you recall if any reps are service multiple roles (i.e. they're both a claims agent and an attorney)?

Reply:
There aren't any real records with more than 1 type. When I asked GCLAWS about this a long time ago, they said if someone were to have multiple accreditation types active at the same time then they'd get a new registration number and poa code for each one.

If they get a new registration number, then they'd get a new `Veteran::Service::Representative` record.

Reply:
How does all that sound @holdenhinkle ?

Reply:
Perfect, thank you both. That sounds good to me.